### DISCUSSION TITLE
w/ PDFHandler on wikimedia1.38 [thumbnails won't work]

### DISCUSSION THREAD

**Discussion by djl314** [ImageMagick/ImageMagick#5729] [Help]
noob here - I compiled and installed on ubtunu running apache2 php7.4 MediaWiki 138 w/ Extension:PDFHandler

These were added to my local settings.php:
$wgUseImageMagick = true;
$wgPdfProcessor = "/usr/bin/gs";
$wgPdfPostProcessor = "/usr/local/bin/convert";
$wgPdfInfo = "/usr/bin/pdfinfo";
$wgPdftoText = "/usr/bin/pdftotext";

I am getting the following error in WikiMedia when uploading PDF files.

Error creating thumbnail: convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/741. convert: no images defined `/tmp/transform_cb79f2dd030d.jpg' @ error/convert.c/ConvertImageCommand/3342

make check returned 87 out of 87 pass
Testsuite summary for ImageMagick 7.1.0-52
============================================================================
# TOTAL: 87
# PASS:  87
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0

??
DJL314

**Comment by snibgo** [comment]
What version of IM? What is the text output from "`/usr/local/bin/convert -version`"? Are you using IMagick?

A possible reason might be that IM is trying to read a PDF, but failing to execute Ghostscript. To test that:
```
/usr/local/bin/convert somefile.pdf[0] out.png
```
Does this command succeed?

**Reply by djl314** [reply]
failed - same output - here is the output text - I'll investigate GS next

Version: ImageMagick 7.1.0-52 Q16-HDRI x86_64 20549 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): 
Compiler: gcc (9.4)
>>/var/www/wiki$ /usr/local/bin/convert check.pdf out.png
convert: no decode delegate for this image format `' @ error/constitute.c/ReadImage/741.
convert: no images defined `out.png' @ error/convert.c/ConvertImageCommand/3342.

**Comment by fmw42** [comment]
Your install has no delegates for image types. How did you install?  Looks like from source.  You need to install from a binary. See https://imagemagick.org/script/download.php

**Reply by djl314** [reply]
I did install from source - following instructions on this page:  https://www.imagemagick.org/script/install-source.php

I tried to run the rpm's and install from binary but my ubuntu system is missing dependencies and the epel instructions are for redhat - I'm a noob using ubuntu --- still stumbling here.

Any additional advice?

**Reply by djl314** [reply]
its' not Ghostscript - that works

So Im stuck at setting delegates for image types.

**Comment by fmw42** [comment]
I am not an expert on Linux or Ubuntu, but search Google. For example https://askubuntu.com/questions/1042436/how-to-install-delegate-libraries-for-image-magick-7-0-7

**Reply by djl314** [reply]
For anyone reading this - the solution was correctly installing the missing library - I found instructions here:
https://gist.github.com/nickferrando/fb0a44d707c8c3efd92dedd0f79d2911
